cip/1.accepted/CIP2015-09-16-public-type-system-type-annotation.adoc

```diff
@@ -128,17 +128,24 @@ a nullable variant:
 ** Lists (ordered sequences with random access) of elements of a given material or nullable type `T`. The syntax for a nullable lists of elements of type `T` is `LIST? OF T`. Note that accessing a list by index always yields a value of type `T?` since indexing out of bounds is defined to return `null`.
 ** `LIST OF T2` is a subtype of `LIST OF T1` if `T2` is a subtype of `T1`. This is a valid subtyping rule since values in Cypher are immutable. Adding an element of type `T` to a `LIST OF S` would produce a new list of type `LIST OF R`, where `R` is the nearest common supertype of `T` and `S`
 * `MAP`
-** Maps from string keys to values of any type, i.e. `{ name: 'Svensson' }`. The type of map values is `ANY?` since it is unknown to the type system if a given map contains a certain key or not. Note also that maps do not distinguish between missing keys and keys that map to a `null` value. `MAP` is a parent to all property containers, like `NODE` and `RELATIONSHIP`.
+** Maps from string keys to values of any type, i.e. `{ name: 'Svensson' }`.
+   The type of map values is `ANY?` since it is unknown to the type system if a given map contains a certain key or not.
+   Note also that maps distinguish between missing keys and keys that map to a `null` value.
+   As an example, that means that `{name: null}` is not equal to `{}`, the empty map.
+   A use case where this distinction is particularly important is when setting multiple properties on a node.
+   `SET n += {name: null, age: 20}` will remove any existing `name` properties while `SET n += {age: 20}` would leave `name` properties untouched.
 
 ==== Graph Types
 
 The following graph types should be supported by any implementation of Cypher, both in the given material and
 a nullable variant:
 
 * `NODE`
-** A `NODE` is a node in the property graph model with properties, labels, and relationships. A node is an entity and a property container in the model and therefore also a `MAP`.
+** A `NODE` is a node in the property graph model with properties, labels, and relationships.
+   A node is an entity and a property container.
 * `RELATIONSHIP`
-** A `RELATIONSHIP` is a relationship in the property graph model with properties, relationship type, a start node, and an end node. A relationship is an entity and a property container in the model and therefore also a `MAP`.
+** A `RELATIONSHIP` is a relationship in the property graph model with properties, relationship type, a start node, and an end node.
+   A relationship is an entity and a property container.
```

Comment:
I love that you just took that out. I am in favour. I do not know if this particular change has any repercussions, but I don't think so. It would be up to the implementation(s) to find out where and when it would not be fine to treat elements/entities as maps.